### PR TITLE
web: sidebar responds to build progress

### PR DESCRIPTION
**Pull request by yuindustries**
- use SVG for indicator dots (using the text dot results in inconsistent sizes based on user presets)
- CSS spinning animation when building

### CODE REVIEW

File: web/src/Sidebar.tsx [hunk 1]
```diff
@@ -76,21 +83,26 @@ class Sidebar extends PureComponent<SidebarProps> {
         analyticsKey = "ui.interactions.errors"
         link += "/errors"
       }
-      let classes = `resLink resLink--${item.status}`
+
+      let formatter = buildFormatter(enStrings)
+      let hasBuilt = !isZeroTime(item.lastDeployTime)
+      let willBuild = !isZeroTime(item.pendingBuildSince)
```
Comment (yuindustries) [inline web/src/Sidebar.tsx]: Actually, we should show this spinner when we're actively building a resource, not (just?) when there's a pending build.

Comment (maiamcc) [reply web/src/Sidebar.tsx]: i think so yeah!

File: web/src/Sidebar.tsx [hunk 2]
```diff
@@ -12,6 +14,10 @@ import buildFormatter from "react-timeago/lib/formatters/buildFormatter"
 import { isZeroTime } from "./time"
 import PathBuilder from "./PathBuilder"
 
+/*
+We need to grab whether a resource has any pending build
```
Comment (maiamcc) [inline web/src/Sidebar.tsx]: is this a TODO, or...? if it's a code comment, idk what it's referring to

File: web/src/Sidebar.scss [hunk 3]
```diff
@@ -71,18 +66,61 @@
...
+  background-image: linear-gradient(
+    to right,
+    transparent 0%,
+    rgba($color-gray-darkest, $translucent) 20%,
+    transparent 80%
+  );
+  animation: shimmer 1s infinite;
+  animation-timing-function: linear;
+}
+.resLink--building .resLink-icon {
```
Comment (maiamcc) [inline web/src/Sidebar.scss]: niiiiice!

File: web/src/Sidebar.scss [hunk 4]
```diff
@@ -71,18 +66,61 @@
...
   content: "┌";
   color: $color-gray-light;
-  margin-top: -8px;
-}
-.resLink--ok::before {
-  color: $color-green;
-}
-.resLink--pending::before {
-  color: $color-gray-light;
+  margin-top: -8px; // Visually center
```
Comment (nicks) [inline web/src/Sidebar.scss]: $spacing-unit / 4?

File: web/src/Sidebar.tsx [hunk 5]
```diff
@@ -76,21 +82,26 @@ class Sidebar extends PureComponent<SidebarProps> {
...
         link += "/errors"
       }
-      let classes = `resLink resLink--${item.status}`
+
+      let formatter = buildFormatter(enStrings)
+      let hasBuilt = !isZeroTime(item.lastDeployTime)
+      let willBuild = !isZeroTime(item.pendingBuildSince)
+      let timeAgo = <TimeAgo date={item.lastDeployTime} formatter={formatter} />
+
+      let classes = `resLink resLink--${willBuild ? "building" : item.status}`
```
Comment (nicks) [inline web/src/Sidebar.tsx]: should we add a `building` status?

File: web/package.json [hunk 6]
```diff
@@ -29,6 +29,7 @@
     "react-select": "^2.4.2",
     "react-test-renderer": "^16.7.0",
     "react-timeago": "^4.4.0",
+    "styled-components": "^4.2.0",
```
Comment (nicks) [inline web/package.json]: did you actually use this anywhere? i couldn't find it...

Comment (yuindustries) [reply web/package.json]: Thanks for catching this! Must have been a dep I missed out when I uninstalled the spinner kit that DM added.

**Pull request merged**